### PR TITLE
CLI publishing guide for w3

### LINKED ISSUE #1513: Create Publish to IPFS with a CLI quickstart guide + docs review
## Background

Create a version of https://github.com/ipfs/ipfs-docs/pull/1490 (once merged), which uses the CLI to upload to a pinning service.

When doing this, we should make the "content-addressed representation" explicit with the local creation of the CAR file and a CAR file upload (which isn't supported by all pinning services). 

IPFS doesn't have a standard data onboarding endpoint, so there isn't one CLI that can be used to "upload" files to IPFS. Therefore, we should make an informed decision while advising users of alternatives.

### CAR uploads 

> Generally, before data is uploaded, it is locally converted into a CAR file, which contains your data transformed into a format that IPFS can consume (a directed acyclic graph, or [DAG](https://dag.ipfs.io/)). This is very powerful - in this process, a [content address](https://docs.ipfs.tech/concepts/content-addressing/) is locally generated for your data, giving you a guarantee of your data’s fingerprint. Further, you can construct the DAG in useful ways, like making individual files within your folders or key-value pairs in your JSON have their own content addresses.

CAR uploads are currently supported by the following pinning services:
- Web3.storage
- Filebase
- Kubo RPC (not an option for this guide)
- Infura (soon)

See: https://www.notion.so/protocollabs/wip-IPIP-Data-Onboarding-with-HTTP-POST-4c394b8ebb774f2d87d34466019257fc for an overview of the different ways you can upload files to pinning services

### Upload approach 

For simplicity, use the [`/car` HTTP endpoint](https://web3.storage/docs/how-tos/store/#storing-ipfs-content-archives) of the Web3.storage API (beware that CARs up to 100MB are supported)

## Suggested outline

The same as https://github.com/ipfs/ipfs-docs/pull/1490

### PR DESCRIPTION
**Pull request by TheDiscordian**
# Describe your changes
`docs/quickstart/publish_cli.md`

Merged https://github.com/web3-storage/w3cli/blob/main/README.md with https://github.com/ipfs/ipfs-docs/pull/1490

## Deadlines and priority:
From a convo with discordian: 

- Hard to say what the deadline should be, end of week or some point next week should be fine.

- I think this is p2, it's not mission critical (IMO, all the info is already in the README linked in the PR anyways)

# What issue(s) does this address?

Closes #1513 

## Checklist before requesting a review
- [ ] Passing the beta version of the **Check Markdown links for modified files** check. Action results can be viewed [here](https://github.com/ipfs/ipfs-docs/actions/workflows/action.yml).

## Checklist before merging
- [ ] Passing all required checks (The beta **Check Markdown links for modified files** check is not required)

### CODE REVIEW

**Comment by TheDiscordian** [comment]
BTW "Check Markdown links for modified files / markdown-link-check" failed because it incorrectly received a 403, perhaps Filebase is blocking the probe.

File: docs/quickstart/publish_cli.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+   ```
+
+## Upload and pin a file
+
+1. Create a space for your files, and register it.
+
+   ```
+   w3 space create Pictures
+   w3 space register
+   ```
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:66]: Break apart steps for readbability, copyability

For some reason, the change suggester isn't formatting this properly. I'll add it manually

Comment (ElPaisano) [reply docs/quickstart/publish_cli.md:66]: 1. Create a space for your files.

   ```shell
    w3 space create Pictures
   ```
   
1. Register your files.
  
   ```shell
   w3 space register
   ```

File: docs/quickstart/publish_cli.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+1. Upload your file.
+
+   ```
+   w3 up welcome-to-IPFS.jpg
+   ```
+
+Congratulations, you have successfully **pinned** a file to IPFS! 🎉
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:74]:
```suggestion
Congratulations, you have successfully **pinned** a file to IPFS using the w3 cli! 🎉
```

File: docs/quickstart/publish_cli.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+:::callout
+Data pinned to the IPFS network is public by default and retrievable by anyone. Avoid publishing private data or adequately encrypt it before publishing.
+:::
+
+## Prerequisites
+
+- A unix-like terminal
+- Node.js installed to the terminal environment
+- The [following image](../quickstart/images/welcome-to-IPFS.jpg), downloaded and saved on your computer:
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:42]:
```suggestion
- A unix-like terminal.
- Node.js installed in the terminal environment.
- The [following image](../quickstart/images/welcome-to-IPFS.jpg), downloaded and saved on your computer:
```

File: docs/quickstart/publish_cli.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+![image](../quickstart/images/welcome-to-IPFS.jpg)
+
+## Install and register to w3
+
+1. Install the w3 cli tool.
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:48]:
```suggestion
1. Install the [w3 command line interface](https://github.com/web3-storage/w3cli).
```

File: docs/quickstart/publish_cli.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+   ```
+   npm install -g @web3-storage/w3cli
+   ```
+
+1. Register to w3 using your email.
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:53]:
```suggestion
1. Authorize the w3 tool to act on behalf of the account associated with your email address `<your@email.com>`:
```

File: docs/quickstart/publish_cli.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+   ```
+   w3 authorize your@email.com
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:56]:
```suggestion
   w3 authorize <your@email.com>
```

**Review by ElPaisano — COMMENTED** [review]
Looks good overall, just some nitpicks with steps, and rephrasing some stuff to distinguish it from the copy its based off of

I need to manually add one change in cus the review tool wasn't behaving

File: docs/quickstart/publish_cli.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+   ```
+
+## Upload and pin a file
+
+1. Create a space for your files, and register it.
+
+   ```
+   w3 space create Pictures
+   w3 space register
+   ```
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:66]:
```suggestion
1. Create a space for your files.

   ```shell
    w3 space create Pictures
   ```
   
1. Register your files.
  
   ```shell
   w3 space register
   ```
```

File: docs/quickstart/publish_cli.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
+---
+title: 'Publish a file with IPFS using a pinning service in a CLI'
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:2]:
```suggestion
title: 'Publish using the command line'
```

File: docs/quickstart/publish_cli.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+description: 'Learn how to publish content with IPFS by pinning a file to a pinning service using a CLI.'
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:3]:
```suggestion
description: 'Learn how to publish content with IPFS by pinning a file to a pinning service using the command line.'
```

File: docs/quickstart/publish_cli.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+---
+
+# Publish a file with IPFS
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:6]:
```suggestion
# Publish a file with IPFS using the command line
```

File: docs/quickstart/publish_cli.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+   w3 authorize <your@email.com>
+   ```
+
+## Upload and pin a file
+
+1. Create a space for your files.
+
+   ```shell
+    w3 space create Pictures
+
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:65]:
```suggestion

   ```shell
    w3 space create Pictures
   ```
   
```

File: docs/quickstart/publish_cli.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+- The [following image](../quickstart/images/welcome-to-IPFS.jpg), downloaded and saved on your computer:
+
+![image](../quickstart/images/welcome-to-IPFS.jpg)
+
+## Install and register to w3
+
+1. Install the [w3 command line interface](https://github.com/web3-storage/w3cli).
+   ```
+   npm install -g @web3-storage/w3cli
+   ```
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:51]:
```suggestion

   ```shell
   npm install -g @web3-storage/w3cli
   ```
```

File: docs/quickstart/publish_cli.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+1. Authorize the w3 tool to act on behalf of the account associated with your email address `<your@email.com>`:
+
+   ```
+   w3 authorize <your@email.com>
+   ```
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:57]:
```suggestion
   ```shell
   w3 authorize <your@email.com>
   ```
```

File: docs/quickstart/publish_cli.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+## Upload and pin a file
+
+1. Create a space for your files.
+
+   ```shell
+    w3 space create Pictures
+
+1. Upload your file.
+
+   ```
```
Comment (ElPaisano) [inline docs/quickstart/publish_cli.md:68]:
```suggestion
   ```shell
```

**Review by ElPaisano — COMMENTED** [review]
Titles, formatting, nits, fix a few issues my first edit introduced 🤭

File: docs/quickstart/publish_cli.md (hunk 1 above)
Comment (ElPaisano) [reply docs/quickstart/publish_cli.md]: added in https://github.com/ipfs/ipfs-docs/pull/1548/commits/3c6a6fb5f71bfc5e41f7609bfd5cf73f1cc22367

**Review by ElPaisano — CHANGES_REQUESTED** [review]
In my initial passes, missed the fact that this is missing a menu entry in the config file. That needs to be added in or this won't be visible / accessible from the menu

**Comment by ElPaisano** [comment]
@TheDiscordian I merged all of this into https://github.com/ipfs/ipfs-docs/pull/1551 to finish up the remaining edits. I'll tag you as reviewer on #1551 so that you can sanity check my updates.

**Pull request merged**